optuna/integration/catalyst.py

```diff
@@ -1,49 +1,38 @@
-from typing import Any
+from packaging import version
 
-import optuna
-from optuna._experimental import experimental
+from optuna._deprecated import deprecated
 from optuna._imports import try_import
 
 
 with try_import() as _imports:
-    from catalyst.dl import Callback
+    import catalyst
+
+    if version.parse(catalyst.__version__) < version.parse("21.3"):
+        raise ImportError(
+            f"You don't have Catalyst 21.3 installed! Catalyst version: {catalyst.__version__}"
+        )
+    from catalyst.dl import OptunaPruningCallback
 
 if not _imports.is_successful():
-    Callback = object  # NOQA
+    OptunaPruningCallback = object  # NOQA
 
 
-@experimental("2.0.0")
-class CatalystPruningCallback(Callback):
+@deprecated("2.7.0")
+class CatalystPruningCallback(OptunaPruningCallback):  # type: ignore
     """Catalyst callback to prune unpromising trials.
 
-    See `the example <https://github.com/optuna/optuna/blob/master/
-    examples/catalyst_simple.py>`_ if you want to add a pruning callback
-    which observes the accuracy of Catalyst's ``SupervisedRunner``.
+
+    .. seealso::
+
+        See https://catalyst-team.github.io/catalyst/api/callbacks.html?highlight=optuna#catalyst.callbacks.optuna.OptunaPruningCallback
+        for the detailed documentation.
 
     Args:
-        trial:
-            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
-            objective function.
-        metric (str):
-            Name of a metric, which is passed to `catalyst.core.State.valid_metrics` dictionary to
-            fetch the value of metric computed on validation set. Pruning decision is made based
-            on this value.
-    """
-
-    def __init__(self, trial: optuna.trial.Trial, metric: str = "loss") -> None:
-
-        # set order=1000 to run pruning callback after other callbacks
-        # refer to `catalyst.core.CallbackOrder`
-        _imports.check()
-        super().__init__(order=1000)
-
-        self._trial = trial
-        self.metric = metric
-
-    def on_epoch_end(self, state: Any) -> None:
-
-        current_score = state.valid_metrics[self.metric]
-        self._trial.report(current_score, state.epoch)
-        if self._trial.should_prune():
-            message = "Trial was pruned at epoch {}.".format(state.epoch)
-            raise optuna.TrialPruned(message)
+        loader_key: loader key for best model selection (based on metric score over the dataset)
+        metric_key: metric key for best model selection (based on metric score over the dataset)
+        minimize: boolean flag to minimize the required metric
+        min_delta: minimal delta for metric improve
+        trial:  Optuna.Trial for experiment.
```

Comment:
This class is just a wrapper of Catalyst's `OptunaPruningCallback`, and it does not have any logic. How about removing the arguments and just guide to the Catalyst's reference? The arguments may be updated by Catalyst, but we may not be able to notice the change.